app/models/health.rb

```diff
@@ -0,0 +1,98 @@
+class Health
+  # things to check: delayed_job, background crons, database, ECHO, CMR, URS, Opensearch, browse-scaler
+
+  def initialize
+    @ok = true
+  end
+
+  def edsc_status
+    {ok?: @ok}
+  end
+
+  def delayed_job_status
+    # The job hasn't been run yet. Go on to check retrieval.created_at. If the time is more than 10 minutes (threshold)
+    # ago, consider the delayed job is down.
+    job = DelayedJob.last
+    retrieval = Retrieval.find_by_id(job.id)
+
+    if job.run_at.nil? && retrieval.created_at < 10.minutes.ago
```

Comment:
I am more inclined to your solution. But the condition needs slightly changed to avoid this false negative situation: if a retrieval is saved one minute ago and the delayed job on that one hasn't started yet, the query will return (at least) the latest entry. I haven't figured out how to make it better though. I will send out an update shortly.